### ISSUE TITLE
Nesting some BBCode throws HTTP ERROR 500 | PHP Fatal error

### ISSUE THREAD

**Issue by donvercety** [punbb/punbb#160]
**My setup:**
- lighttpd/1.4.53
- PHP: 7.3.29-1~deb10u1
- SQLite3 3.27.2
- PunBB 1.4.6

**My Setup 2:**
- PHP 7.2.24-0ubuntu0.18.04.8 with the build in PHP server

**Reproduce:**
Just use the BBCode example in any post, than hit submit or preview.
```
[list][*]Example list item 1.[/*][*]Example list item 2.[/*][*]Example list item 3.[/*][/list]
```

Or this:
```
[list]
[*]Test[/*]
[/list]
```

This works as expected:
```
[b][url]http://github.com[/url][/b]
```

So the problem is not for all nesting. It looks like is related to the `LIST` tag.

**Thrown Error in error log:**
```
2021-08-26 16:52:31: (mod_fastcgi.c.421) FastCGI-stderr: PHP message: PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function {closure}(), 1 passed and exactly 2 expected in /var/www/punbb/include/parser.php:58
2021-08-26 16:52:31: (mod_fastcgi.c.421) FastCGI-stderr: Stack trace:
2021-08-26 16:52:31: (mod_fastcgi.c.421) FastCGI-stderr: #0 [internal function]: {closure}(Array)
2021-08-26 16:52:31: (mod_fastcgi.c.421) FastCGI-stderr: #1 /var/www/punbb/include/parser.php(60): preg_replace_callback('%\\[list(?:=([1a...', Object(Closure), '[list]\n[*]Test[...')
2021-08-26 16:52:31: (mod_fastcgi.c.421) FastCGI-stderr: #2 /var/www/punbb/post.php(179): preparse_bbcode('[list]\n[*]Test[...', Array)
2021-08-26 16:52:31: (mod_fastcgi.c.421) FastCGI-stderr: #3 {main}
2021-08-26 16:52:31: (mod_fastcgi.c.421) FastCGI-stderr:   thrown in /var/www/punbb/include/parser.php on line 58
```

**Comment by MioVisman** [comment]
The author of versions 1.4.5 and 1.4.6 is unknown. This is probably someone from https://www.informer.com/ because only https://punbb.informer.com/ has information on these punbb versions. 
On the forum https://punbb.informer.com/forums/ there is also no activity of the author of versions 1.4.5 and 1.4.6. 

You can use punbb 1.4.4 with my fixes and patches (with support PHP 5.3-8.0):
post https://punbb.informer.com/forums/post/158390/#p158390 and below on the topic.

**Comment by donvercety** [comment]
> You can use punbb 1.4.4 with my fixes and patches (with support PHP 5.3-8.0):
> post https://punbb.informer.com/forums/post/158390/#p158390 and below on the topic.

Using your version fixes the issue, :+1:  thanks!  

For what I see MioVisman version of Punbb is more updated than the current Punbb here.

I will close this issue, coz I don't see any activity here since 2017.
